x-pack/plugins/maps/public/classes/sources/es_search_source/es_search_source.tsx

```diff
@@ -392,6 +395,30 @@ export class ESSearchSource extends AbstractESSource implements ITiledSingleLaye
     return !!(scalingType === SCALING_TYPES.TOP_HITS && topHitsSplitField);
   }
 
+  async isEditable(): Promise<boolean> {
+    if (!getMapAppConfig().enableDrawingFeature) {
+      this._isEditable = false;
+      return this._isEditable;
+    }
+    if (this._isEditable === undefined) {
+      if (!this.indexPattern) {
+        await this.getIndexPattern();
+      }
+
+      const { matchingIndexes } = await getMatchingIndexes(this.indexPattern!.title);
+      this._isEditable =
```

Comment:
How about storing this isEditable state in layerDescriptor.__dataRequest. These instances to not last very long and get recreated any time any layerDescriptor in the map changes. It would be nice to only look this up once per map render instead of each time redux state changes.

Reply:
Hmm, I follow you on putting this somewhere where it's not constantly recreated, but I need to double-check my understanding of adding it to `layerDescriptor.__dataRequests`. So I would update `DataRequestDescriptor` to have a field along the lines of `dataEditable`, but then `layerDescriptor.__dataRequests` is an array, so am I adding a default value to all array elements based on the source criteria I'm using for `_isEditable` now?

Reply:
> so am I adding a default value for dataEditable to all array elements based on the source criteria I'm using for _isEditable now?

Yes, just like the source request or join request, you would add a isEditable data request. The first time syncData is called, VectorLayer would call "this.getSource().isEditable()" and store the results in the isEditable data request. Each other time syncData is called, VectorLayer would find the previous isEditable data request and skip re-fetching again. Then VectorLayer.isEditable would just return false if there is no isEditable data request or the results of isEditable data request.

dataRequests are where async state is fetched/cached.